index.spec.js

```diff
@@ -407,11 +447,121 @@ describe('AppAuth', () => {
         null,
         null,
         false,
+        null,
         true,
         true
       );
     });
 
+    describe('iOS-specific', () => {
+      beforeEach(() => {
+        require('react-native').Platform.OS = 'ios';
+      });
+
+      describe('additionalHeaders parameter', () => {
+        it('calls the native wrapper with additional headers', () => {
+          const additionalHeaders = { header: 'a-value' };
+          authorize({ ...config, additionalHeaders });
+          expect(mockAuthorize).toHaveBeenCalledWith(
+            config.issuer,
+            config.redirectUrl,
+            config.clientId,
+            config.clientSecret,
+            config.scopes,
+            config.additionalParameters,
+            config.serviceConfiguration,
+            config.skipCodeExchange,
+            additionalHeaders,
+            config.useNonce,
+            config.usePKCE
+          );
+        });
+
+        it('throws an error when values are not Record<string,string>', () => {
+          expect(() => {
+            authorize({
+              ...config,
+              additionalHeaders: {
+                notString: {},
+              },
+            });
+          }).toThrow();
+        });
+      });
+
+      describe('useNonce parameter', () => {
```

Comment:
The tests for _nonce_ and _usePKCE_ has been grouped into the _authorize_ tests, as they were previously put amongst the _refresh_ tests.